Source/Shaders/BillboardCollectionFS.glsl

```diff
@@ -55,4 +64,39 @@ void main()
 #endif
 
     czm_writeLogDepth();
+
+#ifdef CLAMP_TO_GROUND
+    if (v_eyeDepth > -2000.0) {
+        vec2 adjustedST = v_textureCoordinates - v_textureOffset.xy;
+        adjustedST = adjustedST / (v_textureOffset.z - v_textureOffset.x, v_textureOffset.w - v_textureOffset.y);
+
+        vec2 st1 = ((v_dimensions.xy * (v_depthLookupTextureCoordinate1 - adjustedST)) + gl_FragCoord.xy) / czm_viewport.zw;
+        float logDepthOrDepth = czm_unpackDepth(texture2D(czm_globeDepthTexture, st1));
+        vec4 eyeCoordinate = czm_windowToEyeCoordinates(gl_FragCoord.xy, logDepthOrDepth);
+        float globeDepth1 = eyeCoordinate.z / eyeCoordinate.w;
+
+        // negative values go into the screen
+        if (globeDepth1 > v_eyeDepth + czm_epsilon5)
+        {
+            vec2 st2 = ((v_dimensions.xy * (v_depthLookupTextureCoordinate2 - adjustedST)) + gl_FragCoord.xy) / czm_viewport.zw;
+            logDepthOrDepth = czm_unpackDepth(texture2D(czm_globeDepthTexture, st2));
+            eyeCoordinate = czm_windowToEyeCoordinates(gl_FragCoord.xy, logDepthOrDepth);
+            float globeDepth2 = eyeCoordinate.z / eyeCoordinate.w;
+
+            if (globeDepth2 > v_eyeDepth + czm_epsilon5)
```

Comment:
One would think that all of these `if`s would increase performance for fragments that don't execute them, but, unfortunately, that's not the case. It would look cleaner if you compute all of the globe depths and then have one `if` combining all four conditionals for the discard.